dev_guide/app_tutorials/jenkins_pipeline.adoc

```diff
@@ -0,0 +1,185 @@
+
+// timeout in 20 minutes if the pipeline hasn't completed
+timeout(time: 20, unit: 'MINUTES') {
+  // use a nodejs jenkins slave
+  node('nodejs') {
+    // select the default cluster
+    openshift.withCluster() {
+      // select the default project
+      openshift.withProject() {
+        echo "Using project: ${openshift.project()}"
+        stage('clean') {
+          // delete everything with this template label
+          openshift.selector("all", [ template : templateName ]).delete()
+          // delete any secrets with this template label
+          if (openshift.selector("secrets", templateName).exists()) {
+            openshift.selector("secrets", templateName).delete()
+          }
+        }
+        stage('create') {
+          // create a new application from the templatePath
+          openshift.newApp(templatePath)
+        }
+        stage('build') {
+          def builds = openshift.selector("bc", templateName).related('builds')
+          // wait up to 5 minutes for the build to complete
+          timeout(5) {
+            builds.untilEach(1) {
+              return (it.object().status.phase == "Complete")
+            }
+          }
+        }
+        stage('deploy') {
+          def rm = openshift.selector("dc", templateName).rollout()
+          // wait up to 5 minutes for the deployment to complete
+          timeout(5) {
+            openshift.selector("dc", templateName).related('pods').untilEach(1) {
+              return (it.object().status.phase == "Running")
+            }
+          }
+        }
+        stage('tag') {
+          // if everything else succeeded, tag the ${templateName}:latest image as ${templateName}:stage
+          // a pipeline build config for the staging environment can watch for the ${templateName}:stage
+          // image to change and then deploy it to the staging environment
+          openshift.tag("${templateName}:latest", "${templateName}:stage")
+        }
+      }
+    }
+  }
+}
+----
+
+[[creating-the-pipeline]]
+
+== Creating the Pipeline
+
+Now we can create the build config in our OpenShift cluster with the following command:
+
+[source]
+----
+$ oc create -f nodejs-sample-pipeline.yaml
+----
+
+If you don't want to create your own file, you can use the sample from the Origin repository with the following command:
+
+[source]
+----
+$ oc create -f https://raw.githubusercontent.com/openshift/origin/master/examples/jenkins/pipeline/nodejs-sample-pipeline.yaml
+----
+
+For more information about the OpenShift DSL syntax used here see https://jenkins.io/doc/pipeline/steps/openshift-pipeline/[OpenShift Pipeline Steps].
+
+[[starting-the-pipeline]]
+
+== Starting the Pipeline
+
+We can now start the pipeline with the following command:
+
+[source]
+----
+$ oc start-build nodejs-sample-pipeline
+----
+
+Once the pipeline has started you should see the following actions performed within your project:
+
+ * a node.js Jenkins slave pod will be created
```

Comment:
even before this, the jenkins server in your project will kick off a job instance.

Reply:
updated in next push (also fixed the pipeline to spin up a node.js slave pod again, had lost it when switch to the declarative syntax)